gef-ui/server/utils/helpers.js

```diff
@@ -297,25 +482,69 @@ const futureDateInRange = ({ day, month, year }, days) => {
      * 86400000 = 24 hours * 60 minutes * 60 seconds * 1000 ms
      * Number of ms in a day
      * */
-    range += (86400000 * days);
+    range += 86400000 * days;
     return input <= range;
   }
   return false;
 };
 
-const coverDatesConfirmed = (facilities) => (
-  facilities.items.filter(({ details }) => details.hasBeenIssued).length === facilities.items.filter(({ details }) => details.coverDateConfirmed).length
-);
+const coverDatesConfirmed = (facilities) =>
+  facilities.items.filter(({ details }) => details.hasBeenIssued).length === facilities.items.filter(({ details }) => details.coverDateConfirmed).length;
+
+/*
+   function returns true or false based on length of array
+   of facilities that have changed to issued from unissued
+*/
+const hasChangedToIssued = (application) => {
+  const changedToIssued = facilitiesChangedToIssuedAsArray(application);
+
+  return changedToIssued.length > 0;
+};
 
 const makerCanReSubmit = (maker, application) => {
   const acceptableStatus = [
     CONSTANTS.DEAL_STATUS.UKEF_APPROVED_WITHOUT_CONDITIONS,
     CONSTANTS.DEAL_STATUS.UKEF_APPROVED_WITH_CONDITIONS,
+    CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED,
   ];
+  let facilitiesChangedToIssued = true;
+
+  // only if AIN -> ensures a facility has changed to issued before resubmitting to bank
+  if (application.status === CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED) {
+    facilitiesChangedToIssued = hasChangedToIssued(application);
+  }
   const coverDateConfirmed = coverDatesConfirmed(application.facilities);
   const makerAuthorised = (maker.roles.includes('maker') && maker.bank.id === application.bankId);
 
-  return (coverDateConfirmed && acceptableStatus.includes(application.status) && makerAuthorised);
+  return coverDateConfirmed && facilitiesChangedToIssued && acceptableStatus.includes(application.status) && makerAuthorised;
+};
+
+/**
+ * checks if any comments present
+ * has extra clause for if app.comments as some e2e tests do not contain this element
+ * stops crashing
+ * */
+
```

Comment:
We could just remove this comment, the code explains itself

Reply:
done